src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,25 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (names.length == 0 || data.length == 0 || dateFrom.isEmpty() || dateTo.isEmpty()) {
+            return null;
+        }
+        StringBuilder results = new StringBuilder("Report for period ");
+        int firstIndex = dateFrom.indexOf('=') + 3;
+        results.append(dateFrom.substring(firstIndex, dateFrom.lastIndexOf(dateFrom)))
```

Comment:
The 'lastIndexOf' method is incorrectly used here. It should find the last index of a specified character or string, not the entire string itself.